src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs

```diff
@@ -268,12 +268,7 @@ public ViewViewComponentResult View<TModel>(TModel model)
         /// <returns>A <see cref="ViewViewComponentResult"/>.</returns>
         public ViewViewComponentResult View<TModel>(string viewName, TModel model)
         {
-            if (model != null)
-            {
-                ViewData.Model = model;
-            }
-
-            var viewData = new ViewDataDictionary<TModel>(ViewData);
+            var viewData = new ViewDataDictionary<TModel>(ViewData, model);
```

Comment:
E.g. in the `InvokeAsync()` method
``` c#
var result = View(new MyShinyNewModel());
ViewData.Model.SomeShinyProperty = "hello world";
return result;
```
With the previous implementation and in the equivalent action, the above works. With the new code, `ViewData.Model` will be the old value and not necessarily a `MyShinyNewModel` instance.

As I said, this isn't at all common. The question to @Eilon is whether it matters enough to set `ViewData.Model` separately.